python/lsst/daf/butler/registry/_registry.py

```diff
@@ -151,35 +171,16 @@ def createFromConfig(cls, config: Optional[Union[RegistryConfig, str]] = None,
         registry : `Registry`
             A new `Registry` instance.
         """
-        if isinstance(config, str):
-            config = RegistryConfig(config)
-        elif config is None:
-            config = RegistryConfig()
-        elif not isinstance(config, RegistryConfig):
-            raise TypeError(f"Incompatible Registry configuration type: {type(config)}")
-        config.replaceRoot(butlerRoot)
-
-        if isinstance(dimensionConfig, str):
-            dimensionConfig = DimensionConfig(config)
-        elif dimensionConfig is None:
-            dimensionConfig = DimensionConfig()
-        elif not isinstance(dimensionConfig, DimensionConfig):
-            raise TypeError(f"Incompatible Dimension configuration type: {type(dimensionConfig)}")
-
-        DatabaseClass = config.getDatabaseClass()
-        database = DatabaseClass.fromUri(str(config.connectionString), origin=config.get("origin", 0),
-                                         namespace=config.get("namespace"))
-        managerTypes = RegistryManagerTypes.fromConfig(config)
-        managers = managerTypes.makeRepo(database, dimensionConfig)
-        return cls(database, RegistryDefaults(), managers)
+        registry_cls, registry_config = cls.determineTrampoline(config)
+        return registry_cls.createFromConfig(config, dimensionConfig, butlerRoot)
```

Comment:
Yes, thanks. It wouldn't matter in this particular case because the subclass forces it to a `RegistryConfig` anyhow but it definitely was something I should have done.